### PR TITLE
Introduces Gene Shears to Botany, and several other QOL tweaks.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
If you or a loved one played pre-fix hydroponics in the last 24 hours, you aren't entitled to compensation, but I will loudly apologize to you.

(I swear this one isn't a speedmerge PR.)
##  About The Pull Request

In my infinite wisdom, I started work on a chemical that would allow for incremental changes to a plant's chemical genes, as well as their traits. After a (brief) discussion on discord about why that would be the **_fucking worst idea_** ever due to how much RNG it would introduce, this idea will work instead, which are the **_Botanogenetic Plant Shears_**, BP Shears, or Gene Shears. These can be used on a plant, at the cost of plant health, similar to plant grafts, 

This can be used to remove chemical traits, as well as physical traits, enabling you to regain control of your plant's genes and traits, as well as enable you to isolate out less desirable chemicals and traits out of your plant. To make BP Shears, you'll need the experimental tools tech from the tech tree. As it's not a roundstart tool, you'll want to use the early parts of your round dedicated to improving stats, or creating rough gene pools for plants that wouldn't 

To make up for the material cost of needing to craft shears, I've also expanded the existing chemical solidification reactions to include uranium. Most chemical solidification reactions are woefully underused, but with this you won't be fully locked out should there be no miners on the station, with some effort that is. Make sure to watch your mixing order!

Hydroponics Trays have a new higher starting reagents tank of 20 units, up from the original 10. This one I could go either way on but from player feedback starting with a higher initial value enables the player to get more out of mixing reagents to adjust plant stats, which was the intended goal.

While tweaking other stuff, I realized that using autogrow needed an extra call of RefreshParts() when turned off, or otherwise it would set itself to the default lower nutrient drain rate forever (Until upgraded again) Additionally, autogrow won't be toggled when pulling around trays.

## Why It's Good For The Game
Working down the changelog:
 *Botany not being able to remove traits was a massive oversight and something that was glossed over about 2 months ago when I started working on this, so this allows high level botany play without it having the same level of power and control the gene modder provided.
 *By adding the uranium solidification reaction it allows for a dynamic option that botanist players can use to benefit the station, ooooor just to produce their own shears if there isn't a miner available.
 *I don't plan on revisiting the chemical idea, as even suggesting using a part of virology's gameplay loop was enough to get a maintainer to make me feel like garbage. Plus it was kind of a lame concept.
 *Encourages a larger part of the rework's gameplay to players, and makes better use of reagents that use quantity to determine their effects.
 *Fixes an annoying bug that gave conflicting feedback.

## Changelog
:cl:
add: Botany can now print shears capable of removing plant genes, after researching the botany node.
add: Uranium can now be solidified in a chemical reaction, similar to gold and plasma.
del: Removed Plant Thinner and Trait Thinner chemicals (Unused)
tweak: Hydroponics trays now have a starting chemical capacity of 20.
fix: Using a beaker on a hydroponics tray while the tray is full no longer acts like it's transferring reagents.
tweak: Ctrl-Clicking a tray will only enable autogrow when it's anchored.
/:cl:

### CODE REVIEW

**Comment by Angustmeta** [comment]
For clarity, do the shears remove a random trait/chem or do you choose one?

File: code/modules/hydroponics/hydroponics.dm [hunk 1]
```diff
@@ -252,10 +252,9 @@
 
 //This is where stability mutations exist now.
 			switch(myseed.instability)
-				if(80 to 100)
-					if(prob(20))
-						mutate(0, 0, 0, 0, 0, 0, 0, 10, 0) //Exceedingly low odds of gaining a trait.
-				if(60 to 79)
+				if(100 to 80)
+					mutate(0, 0, 0, 0, 0, 0, 0, 5, 0) //Exceedingly low odds of gaining a trait.
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/hydroponics/hydroponics.dm:256]: was it intentional to double the probability for mutation? I don't see it documented.

if you were trying to make them the same, you're looking for
```suggestion
					mutate(0, 0, 0, 0, 0, 0, 0, 2, 0) //Exceedingly low odds of gaining a trait.
```

File: code/modules/hydroponics/hydroponics.dm [hunk 2]
```diff
@@ -78,8 +78,8 @@
 	for (var/obj/item/stock_parts/manipulator/M in component_parts)
 		rating = M.rating
 	maxwater = tmp_capacity * 50 // Up to 300
-	maxnutri = tmp_capacity * 5 // Up to 30
-	reagents.maximum_volume = maxnutri
+	maxnutri = (tmp_capacity * 5) +10 // Up to 30
+	reagents.maximum_volume = maxnutri + 10
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/hydroponics/hydroponics.dm:82]: are these magic numbers relative to 20?

File: code/modules/hydroponics/hydroponics.dm [hunk 3]
```diff
@@ -648,7 +649,7 @@
 			to_chat(user, "<span class='notice'>This plant must be harvestable in order to be grafted.</span>")
 			return
 		else if(myseed.grafted)
-			to_chat(user, "<span class='notice'>This plant has already been grafted.</span>")
+			to_chat(user, "<span class='notice'>This plant has already had something sheared or grafted off from it.</span>")
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/hydroponics/hydroponics.dm:652]: I don't see where shears set this to true.

Also will need to update documentation of this variable if not change the name now that it also considers shearing.

File: code/modules/hydroponics/hydroponics_chemreact.dm [hunk 4]
```diff
@@ -7,9 +7,9 @@
 	if(myseed)
 		myseed.on_chem_reaction(reagents) //In case seeds have some special interactions with special chems, currently only used by vines
 	//FINE WE'LL DO IT THIS WAY.
-	for(var/c in reagents.reagent_list)
-		var/datum/reagent/chem = c
-		chem.on_hydroponics_apply(myseed, reagents, src, user)
+	for(var/datum/reagent/chem in reagents.reagent_list)
+		if(chem)
+			chem.on_hydroponics_apply(myseed, reagents, src, user)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/hydroponics/hydroponics_chemreact.dm:12]: what's this meant to do? If you don't have the chem it shouldn't be in the reagent list.

Also this should still be untyped unless you're expecting non-reagents to be here (it shouldn't be)

Comment (ExcessiveUseOfCobblestone) [reply code/modules/hydroponics/hydroponics_chemreact.dm:12]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#istypeless-for-loops

File: code/modules/hydroponics/hydroponics.dm (hunk 3 above)
Comment (ArcaneMusic) [reply code/modules/hydroponics/hydroponics.dm]: I had it so that gene shearing was a once/per life like grafts, but I decided against it by going with just causing plant_health damage instead.

File: code/modules/hydroponics/hydroponics_chemreact.dm (hunk 4 above)
Comment (ArcaneMusic) [reply code/modules/hydroponics/hydroponics_chemreact.dm]: I've legitimately never seen this before, good to know.

**Comment by JJRcop** [comment]
The first PR was not properly documented for dmdoc. dmdoc requires block comments to begin with `/**`, note the extra asterisk, and line comments to begin with `///`. Reading [AUTODOC_GUIDE.md](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md) will help avoid these mistakes.

File: code/modules/hydroponics/hydroponics.dm [hunk 5]
```diff
@@ -768,6 +798,8 @@
 	self_sustaining = !self_sustaining
 	idle_power_usage = self_sustaining ? 5000 : 0
 	to_chat(user, "<span class='notice'>You [self_sustaining ? "activate" : "deactivated"] [src]'s autogrow function[self_sustaining ? ", maintaining the tray's health while using high amounts of power" : ""].")
+	if(!self_sustaining)
+		RefreshParts()
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/hydroponics/hydroponics.dm:802]: why is this here?

Also can you set it so it requires anchoring to switch self_sustain? when I try to move trays I have to use rightclick or else they turn on.

Comment (ArcaneMusic) [reply code/modules/hydroponics/hydroponics.dm:802]: Currently, turning on self-sustaining enables a slower nutrient drain rate at earlier part levels. By not refreshing parts when turned off however, you can enable and re-enable autogrow to give all trays a drain-rate on par with tier 2 parts. Throwing in a RefreshParts is a safety net here.

Comment (ExcessiveUseOfCobblestone) [reply code/modules/hydroponics/hydroponics.dm:820]: shouldn't the safety net be like on process or something?

it looks like the bug would still exist if it switched via another aspect that isn't a user ctrl clicking it.

**Comment by trollbreeder** [comment]
image
Branchname

**Comment by Yttriums** [comment]
Wasn't here for the original PR but I wanted to drop in to say that the DNA manipulator removal is trash and should be reversed.

**Comment by ArcaneMusic** [comment]
> Wasn't here for the original PR but I wanted to drop in to say that the DNA manipulator removal is trash and should be reversed.

The DNA Manipulator is a nice tool to have, but the main issue with it is that outside of plant mutations, it discouraged you from actually growing the vast majority of the plants available to the player. For example, If you wanted liquid contents, you literally only needed to take a tomato seed out of the vendor and shove it into the manipulator to have liquid contents, forever, at no risk whatsoever. It encouraged a really stale gameplay loop of not growing the vast majority of plants whose traits you need for more interesting builds. A large focus of the design of this rework was encouraging players to actually have to fully grow more plants, benefiting the whole service department in the process.

With the existing changes, in combination with this PR's feature, you'll be able to do just the same amount of gene tailoring and crazy batshit plant manipulation as before, but it prevents you from speedrunning the department's endgame in a total of 15 minutes.

**Comment by ExcessiveUseOfCobblestone** [comment]
wouldn't it be a nice qol if trays showed you their nutri ratios if you had science goggles.

image

**Comment by ArcaneMusic** [comment]
> wouldn't it be a nice qol if trays showed you their nutri ratios if you had science goggles.
> 
> image

I'll throw this into the plant analyzer PR to keep less file conflicts from this PR.

**Comment by adamkad1** [comment]
Those should be added to pod people ruin

**Comment by adamkad1** [comment]
Those should be added to pod people ruin

**Comment by adamkad1** [comment]
Those should be added to pod people ruin
